### PR TITLE
Task js DOM

### PR DESCRIPTION
**Pull request by loki13btz**
### side-menu (dom-api-task)

[Demo](https://loki13btz.github.io/side-menu/) |
[Code base](https://github.com/loki13btz/side-menu)

The code is submitted in a dedicated feature branch.

Only code files are submitted.

Please, review.

### CODE REVIEW

**Review by madmaxWMFU — CHANGES_REQUESTED** [review]
Don't use any library. In this task only vanilla JS.

**Comment by loki13btz** [comment]
@madmaxWMFU I have removed the use of jQuery, please take a look;)

File: submissions/loki13btz/dom-api-task/script.js [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+  }
+  function clearActive() {
+    menuItems.forEach(menuItem => {
+      menuItem.classList.remove("active");
+    });
+  }
+
+  menuItems.forEach(menuItem => {
+    menuItem.addEventListener('click', function (e) {
+      let menuItemId = Number(e.currentTarget.dataset.id);
```
Comment (madmaxWMFU) [inline submissions/loki13btz/dom-api-task/script.js:67]: Add listener, not for all items of the menu. Only for the parent (menuItems)

File: submissions/loki13btz/dom-api-task/script.js [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+      clearActive();
+      e.currentTarget.classList.add("active");
+      renderContent(menuItemId);
+    });
+    menuItem.addEventListener('keydown', function (e) {
+      if (e.keyCode == 32) {
+        let menuItemId = Number(e.currentTarget.dataset.id);
+        clearActive();
+        e.currentTarget.classList.add("active");
+        renderContent(menuItemId);
```
Comment (madmaxWMFU) [inline submissions/loki13btz/dom-api-task/script.js:77]: DRY

File: submissions/loki13btz/dom-api-task/script.js [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+      }
+    });
+  });
+  renderContent(1);
```
Comment (madmaxWMFU) [inline submissions/loki13btz/dom-api-task/script.js:81]: Set a value to the initial variable.

File: submissions/loki13btz/dom-api-task/script.js [hunk 4]
```diff
@@ -0,0 +1,82 @@
...
+    id: 5,
+    img: "./img/sea5.jpg",
+    title: "Contact page",
+    content:
+      "Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum.",
+  },
+];
+
+document.addEventListener('DOMContentLoaded', function() {
+  let menuItems = document.querySelectorAll(".side-menu_item");
```
Comment (madmaxWMFU) [inline submissions/loki13btz/dom-api-task/script.js:40]: All variables in your context must be `const`.

File: submissions/loki13btz/dom-api-task/script.js [hunk 5]
```diff
@@ -0,0 +1,82 @@
...
+  {
+    id: 5,
+    img: "./img/sea5.jpg",
+    title: "Contact page",
+    content:
+      "Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum.",
+  },
+];
+
+document.addEventListener('DOMContentLoaded', function() {
```
Comment (madmaxWMFU) [inline submissions/loki13btz/dom-api-task/script.js:39]: Everything else can be moved beyond the body of the function and into separate created functions.
```suggestion
document.addEventListener('DOMContentLoaded', function() {
  renderContent(initialValue);
}
```

**Review by madmaxWMFU — CHANGES_REQUESTED** [review]

File: submissions/loki13btz/dom-api-task/script.js [hunk 6]
```diff
@@ -0,0 +1,91 @@
...
+  menuItemList.addEventListener("click", function (e) {
+    const menuItem = e.target;
+    addActive(menuItem);
+  });
+  menuItemList.addEventListener("keydown", function (e) {
+    const menuItem = e.target;
+    if (e.keyCode == 32) {
+      addActive(menuItem);
+    }
+  });
```
Comment (madmaxWMFU) [inline submissions/loki13btz/dom-api-task/script.js:85]:
```suggestion
  menuItemList.addEventListener("click", function ({ target }) {
    addActive(target);
  });
  menuItemList.addEventListener("keydown", function ({ keyCode, target }) {
    if (keyCode == 32) {
      addActive(target);
    }
  });
```

File: submissions/loki13btz/dom-api-task/script.js [hunk 7]
```diff
@@ -0,0 +1,91 @@
...
+    title: "Contact page",
+    content:
+      "Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum.",
+  },
+];
+const defaultMenuId=1;
+
+function renderContent(menuId) {
+  const menuContent = document.querySelector(".side-menu_content");
+  const pageContent = document.querySelector(".content .content-wrapper");
```
Comment (madmaxWMFU) [inline submissions/loki13btz/dom-api-task/script.js:42]:
```suggestion
  const pageContent = document.querySelector(".content-wrapper");
```

File: submissions/loki13btz/dom-api-task/script.js [hunk 8]
```diff
@@ -0,0 +1,91 @@
...
+  const pageTitle = document.createElement("h1");
+  pageTitle.classList.add("content-title");
+  const pageDescription = document.createElement("p");
+  pageDescription.classList.add("content-description");
+  const { id, img, title, content } = sideMenuContent.find(
+    ({ id }) => id === menuId
+  );
+  pageContent.innerHTML = "";
+  menuContent.style.backgroundImage = `url(${img})`;
```
Comment (madmaxWMFU) [inline submissions/loki13btz/dom-api-task/script.js:51]: Do it by the rules in CSS and class to the element.

**Review by madmaxWMFU — CHANGES_REQUESTED** [review]

**Comment by loki13btz** [comment]
@madmaxWMFU please, take a look the changes

File: submissions/loki13btz/dom-api-task/style.css [hunk 9]
```diff
@@ -0,0 +1,239 @@
...
+}
+
+.content {
+    width: 75%;
+    background: rgba(0, 0, 0, .4);
+}
+
+.content-wrapper {
+    text-align: center;
+    margin-top: 28%;
```
Comment (madmaxWMFU) [inline submissions/loki13btz/dom-api-task/style.css:154]: `padding-top: 28%`
And in line 214 `padding-top: 15%`

File: submissions/loki13btz/dom-api-task/style.css [hunk 10]
```diff
@@ -0,0 +1,239 @@
...
+    color: #fff;
+    font-size: 2em;
+    opacity: 90%;
+    margin: 10px 15px;
+    padding-bottom: 20px;
+
+}
+
+@media(max-width:768px) {
+    .side-menu_content {
```
Comment (madmaxWMFU) [inline submissions/loki13btz/dom-api-task/style.css:176]: Mabey you want to write `#side-menu_content`?

**Review by madmaxWMFU — CHANGES_REQUESTED** [review]

**Review by madmaxWMFU — APPROVED** [review]
Great, I approve of your work. Good luck with future tasks!

**Comment by madmaxWMFU** [comment]
Update your demo!

**Pull request merged**